eclair-core/src/main/scala/fr/acinq/eclair/router/RouteCalculation.scala

```diff
@@ -50,6 +50,18 @@ object RouteCalculation {
   }
 
   def finalizeRoute(d: Data, localNodeId: PublicKey, fr: FinalizeRoute)(implicit ctx: ActorContext, log: DiagnosticLoggingAdapter): Data = {
+    def validateMaxRouteFee(amount: MilliSatoshi, maxFee_opt: Option[MilliSatoshi], hops: Seq[ChannelHop]): Either[Status.Failure, Unit] = {
+      maxFee_opt match {
+        case Some(maxFee) =>
+          val fee = hops.tail.foldLeft(0.msat)((acc, hop) => acc + hop.fee(amount))
+          if (fee > maxFee) {
+            val routeStr = hops.map(hop => s"${hop.shortChannelId}:${hop.fee(amount).toLong}").mkString("->")
```

Comment:
The fee computation is incorrect, since you're going from the sender to the recipient (it should be reversed) and you're including the fee of the first channel, which is likely your own channel so it's not a fee you will be paying. I fixed that and refactored it in https://github.com/ACINQ/eclair/commit/ca293d98d82e22df11821a62aff9fa3e53bf1a3a, feel free to cherry-pick.

Reply:
@rorp ping?